clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerRecord.java

```diff
@@ -85,6 +89,35 @@ public ConsumerRecord(String topic,
                           int serializedValueSize,
                           K key,
                           V value) {
+        this(topic, partition, offset, timestamp, timestampType, checksum, serializedKeySize, serializedValueSize, key, value, new RecordHeaders());
```

Comment:
Saying this, the RecordHeaders uses array list, which does an empty array memory saving, so really the over head for cleaness of code for empty headers is just the RecordHeaders object, no real sizeable data/memory.